golem-worker-executor-base/src/services/rdbms/tests.rs

```diff
@@ -0,0 +1,2938 @@
+                name                text    NOT NULL,
+                tags                text[],
+                created_on          timestamp DEFAULT NOW()
+            );
+        "#;
+
+    rdbms_test(
+        rdbms.clone(),
+        &db_address,
+        RdbmsTest::new(
+            vec![StatementTest::execute_test(
+                create_table_statement,
+                vec![],
+                None,
+            )],
+            None,
+        ),
+    )
+    .await;
+
+    let insert_statement = r#"
+            INSERT INTO test_users
+            (user_id, name, tags)
+            VALUES
+            ($1, $2, $3)
+        "#;
+
+    let count = 60;
+
+    let mut rows: Vec<DbRow<postgres_types::DbValue>> = Vec::with_capacity(count);
+
+    let mut statements: Vec<StatementTest<PostgresType>> = Vec::with_capacity(count);
+
+    for i in 0..count {
+        let params: Vec<postgres_types::DbValue> = vec![
+            postgres_types::DbValue::Text(format!("{:03}", i)),
+            postgres_types::DbValue::Text(format!("name-{:03}", i)),
+            postgres_types::DbValue::Array(vec![
+                postgres_types::DbValue::Text(format!("tag-1-{}", i)),
+                postgres_types::DbValue::Text(format!("tag-2-{}", i)),
+            ]),
+        ];
+
+        statements.push(StatementTest::execute_test(
+            insert_statement,
+            params.clone(),
+            Some(1),
+        ));
+
+        rows.push(DbRow { values: params });
+    }
+
+    let select_statement_test = StatementTest::query_test(
+        "SELECT user_id, name, tags FROM test_users ORDER BY user_id ASC",
+        vec![],
+        None,
+        Some(rows.clone()),
+    );
+
+    statements.push(select_statement_test.clone());
+
+    rdbms_test(
+        rdbms.clone(),
+        &db_address,
+        RdbmsTest::new(statements, Some(TransactionEnd::Commit)),
+    )
+    .await;
+
+    let delete_statement_test = StatementTest::execute_test("DELETE FROM test_users", vec![], None);
+
+    rdbms_test(
+        rdbms.clone(),
+        &db_address,
+        RdbmsTest::new(
+            vec![delete_statement_test.clone()],
+            Some(TransactionEnd::Rollback),
+        ),
+    )
+    .await;
+
+    rdbms_test(
+        rdbms.clone(),
+        &db_address,
+        RdbmsTest::new(
+            vec![select_statement_test.with_query_stream_expected(None, Some(rows.clone()))],
+            Some(TransactionEnd::Commit),
+        ),
+    )
+    .await;
+
+    rdbms_test(
+        rdbms.clone(),
+        &db_address,
+        RdbmsTest::new(vec![delete_statement_test], Some(TransactionEnd::Commit)),
+    )
+    .await;
+
+    rdbms_test(
+        rdbms.clone(),
+        &db_address,
+        RdbmsTest::new(
+            vec![select_statement_test.with_query_expected(Some(vec![]), Some(vec![]))],
+            Some(TransactionEnd::None),
+        ),
+    )
+    .await;
+}
+
+#[test]
+async fn postgres_create_insert_select_test(
```

Comment:
May be these tests are necessary, but I couldn't understand yet why. Could you please explain @justcoon ?

Reply:
those test are testing conversion and type mappings, query execution and transaction implementations